src/dos/drive_virtual.cpp

```diff
@@ -348,11 +348,13 @@ Virtual_File::Virtual_File(const vfile_data_t& in_data)
 
 bool Virtual_File::Read(uint8_t * data,uint16_t * size) {
 	uint32_t left = file_data->size() - file_pos;
+	assert(file_pos <= file_data->size());
+	uint8_t* src  = file_data->data() + file_pos;
 	if (left <= *size) {
-		memcpy(data, &((*file_data)[file_pos]), left);
+		memcpy(data, src, left);
 		*size = (uint16_t)left;
 	} else {
-		memcpy(data, &((*file_data)[file_pos]), *size);
+		memcpy(data, src, *size);
 	}
 	file_pos += *size;
 	return true;
```

Comment:
The if/else branches are confusing; maybe another commit refactoring to something like:

```c++
bool Virtual_File::Read(uint8_t * data, uint16_t * bytes_requested) {
  assert(file_pos <= file_data->size());
  const uint32_t bytes_left = file_data->size() - file_pos;
 
  const auto num_bytes = std::min(bytes_left, static_cast<uint32_t>(bytes_requested));

  const auto source = file_data->data() + file_pos;

  memcpy(data, source, num_bytes);
}